CMakePresets.json

```diff
@@ -0,0 +1,77 @@
+{
+  "version": 1,
+  "cmakeMinimumRequired": {
+    "major": 3,
+    "minor": 16,
+    "patch": 0
+  },
+  "configurePresets": [
+    {
+      "name": "gcc-debug",
+      "displayName": "GCC (Debug)",
+      "description": "Debug build using Ninja generator and GCC-compatible compiler",
+      "generator": "Ninja",
+      "binaryDir": "${sourceDir}/build",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Debug"
+      }
+    },
+    {
+      "name": "gcc-release",
+      "displayName": "GCC (Release)",
+      "description": "Release build using Ninja generator and GCC-compatible compiler",
+      "generator": "Ninja",
+      "binaryDir": "${sourceDir}/build",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Release",
+        "CMAKE_CXX_FLAGS_RELEASE": "-O2 -DNDEBUG"
```

Comment:
We take pains to ensure that NDEBUG is never defined (e.g. Util.h), so this should really be `-UNDEBUG`

Reply:
This is to match the existing CMake behavior and to help out with #5505. Just removing `-DNDEBUG` should be enough (plus documenting it).